content/articles/how-to-implement-structured-logging-in-adonisjs/index.md

```diff
@@ -0,0 +1,261 @@
+## How to implement Structured Logging in Adonis.js
+Logs are among the essential tools for application developers during the debugging process and for analysts monitoring applications in the production environment.
+
+Typical logs are textual formats that are easily readable by humans but very difficult for machines to read and process.
+
+Assuming you want to use automated processing to investigate log files, or you want to use algorithms to categorize, index, and search through log files based on specific parameters, typical logs will not work as expected.
+
+That is where structured logging becomes useful. Structured logging is a way to record the information in a way that is easy for machines to process.
+
+This article will show the best practice in implementing structured logging in Adonis.js 5, discussing in detail, the best ways involved in implementing a scalable logging system in your RESTFUL API. 
+
+We will build a real-world app to demonstrate how to debug easily with excellent structured logs in Adonis.js.
+
+### Prerequisites
+Before proceeding, you are required to have the following:
+1. An understanding of TypeScript and Node.js.
+2. An [understanding of Adonis.js](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide)
+3. Knowldege of building [APIs with Adonis.js](/engineering-education/build-a-restful-api-with-adonisjs/)
+
+### Goal of the tutorial
+I will be showing you the best practice in handling structured logging in Adonis.js. We will build a real-world Forum app to demonstrate how to boost maintainability in Adonis.js.
+
+In this tutorial, you will learn:
+
+- How to build a highly-maintainable app with Adonis.js.
+- How to implement structured logging in Adonis.js Restful API.
+- How to choose what to log and how to log it.
+- About the concept of structured logging and how to maintain a large app with it.
+- The best practices in building large-scale web apps with Adonis.js.
+
+### Scaffolding an Adonis.js project
+Getting started with Adonis.js is easy because of its good documentation. To learn more about the latest Adonis.js 5, reading through [Adonis.js 5 ultimate guide](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide) and [building a RESTFUL API with Adonis 5](/engineering-education/build-a-restful-api-with-adonisjs/) to get up to speed.
+
+To install Adonis.js 5, run the command below:
+
+```bash
+npm init adonis-ts-app@latest forum-api
+```
+
+The command creates a new Adonis.js 5 project, properly configured and ready to be used. We will continue with the Forum API developed with the article on: [How to Implement Caching using Adonis.js 5 tutorial](/engineering-education/how-to-implement-caching-in-adonisjs-5/). You can [clone the repository](https://github.com/Kaperskyguru/adonisjs-forum-api) for an easy setup.
+
```

Comment:
Does the reader have to read the "How to Implement Caching using Adonis.js 5" to be able to follow along with this tutorial? 

It is very hard to follow along with this tutorial, because:
1. The Forum app is an already built app. And:
2. You have not provided information on how to set it up. 
3. You have not explained what this forum app does.
4. There are several libraries which you are using already. ie. Adonis.js, Pino.js, which make it even more confusing.
5. The amount of code in this tutorial is very huge and not well documented and explained.

So, the readers are editing files that contain existing code. Which they don't know what the code does.

Please consider setting up another simple project from scratch to demonstrate the concept you are demonstrating in this tutorial.

Reply:
The article is more of a series. It follows along with 2 articles I have already written